docker/test/integration/steps/steps.py

```diff
@@ -269,11 +361,117 @@ def step_impl(context, content, file_name, path, seconds):
+    # callbacks to be triggered.
+    producer.flush(10)
+
+@when("{number_of_messages} kafka messages are sent to the topic \"{topic_name}\"")
+def step_impl(context, number_of_messages, topic_name):
+    producer = Producer({"bootstrap.servers": "localhost:29092", "client.id": socket.gethostname()})
+    for i in range(0, int(number_of_messages)):
+        producer.produce(topic_name, str(uuid.uuid4()).encode("utf-8"))
+    producer.flush(10)
+
+@when("a message with content \"{content}\" is published to the \"{topic_name}\" topic with headers \"{semicolon_separated_headers}\"")
+def step_impl(context, content, topic_name, semicolon_separated_headers):
+    # Confluent kafka does not support multiple headers with same key, another API must be used here.
+    headers = []
+    for header in semicolon_separated_headers.split(";"):
+        kv = header.split(":")
+        headers.append((kv[0].strip(), kv[1].strip().encode("utf-8")))
+    producer = KafkaProducer(bootstrap_servers='localhost:29092')
+    future = producer.send(topic_name, content.encode("utf-8"), headers=headers)
+    result = future.get(timeout=60)
 
 @then("a flowfile with the content \"{content}\" is placed in the monitored directory in less than {duration}")
 def step_impl(context, content, duration):
-    context.test.check_for_file_with_content_generated(content, timeparse(duration))
+    context.test.check_for_single_file_with_content_generated(content, timeparse(duration))
 
+@then("at least one flowfile with the content \"{content}\" is placed in the monitored directory in less than {duration}")
+def step_impl(context, content, duration):
+    context.test.check_for_at_least_one_file_with_content_generated(content, timeparse(duration))
+
+@then("{num_flowfiles} flowfiles are placed in the monitored directory in less than {duration}")
+def step_impl(context, num_flowfiles, duration):
+    if num_flowfiles == 0:
+        context.execute_steps("""no files are placed in the monitored directory in {duration} of running time""".
+            format(duration=duration))
+        return
+    context.test.check_for_num_files_generated(int(num_flowfiles), timeparse(duration))
+
+@then("two flowfiles with the contents \"{content_1}\" and \"{content_2}\" are placed in the monitored directory in less than {duration}")
+def step_impl(context, content_1, content_2, duration):
+    wait_start_time = time.perf_counter()
+    context.test.wait_for_multiple_output_files(timeparse(duration), 2)
+    context.execute_steps("""
+        then a flowfile with the content \"{content}\" is placed in the monitored directory in less than {duration}""".
+        format(content=content_1, duration=duration))
+    context.execute_steps("""
+        then a flowfile with the content \"{content}\" is placed in the monitored directory in less than {duration}""".
+        format(content=content_1, duration=str(timeparse(duration) - (time.perf_counter() - wait_start_time)) + " seconds"))
```

Comment:
shouldn't this be `content_2`?

Reply:
Correct, that is an error, I asserted the same thing twice. Updated to perform both checks.